compiler/tests/fail/stack_allocation/x86-64/stack_ptr_in_expr.jazz

```diff
@@ -0,0 +1,9 @@
+export fn main (reg ptr u64[1] r) -> reg u64 {
+  stack ptr u64[1] sp;
+  reg u64 res;
+
+  sp = r;
+  res = sp[0] + 1;
```

Comment:
stack allocation: sp (defined at "tests/fail/stack_allocation/x86-64/stack_ptr_in_expr.jazz", line 5 (2-4)) is a stack pointer, it should not appear in an expression

sp can appear in an expression like in r = sp.

Reply:
```
"tests/fail/stack_allocation/x86-64/stack_ptr_in_expr.jazz", line 6 (8-10):
compilation error in function main:
stack allocation: sp (defined at "tests/fail/stack_allocation/x86-64/stack_ptr_in_expr.jazz", line 5 (2-4)) is a stack ptr, it should not appear in a sub-expression
```